### PR TITLE
Scaffold users

### PR DESCRIPTION
**Pull request by mikeyrayvon**
create, update, and delete users (Authentication) and user profiles (Database/users)
uses https://github.com/interglobalvision/functions-andamiaje-co to call admin functions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
   "private": true,
   "dependencies": {
+    "axios": "^0.17.1",
     "custom-react-scripts": "^0.2.1",
     "draft-js": "^0.10.4",
     "firebase": "^4.8.1",
     "globie-styl-library": "^1.0.0",
     "kouto-swiss": "^1.1.0",
     "moment": "^2.20.1",
+    "random-string": "^0.2.0",
```
Comment (varffs) [inline package.json:13]: Maybe not important considering the scope of the function but looks like a stale project with no commits in 9 months and not many users. Can we not find this function in something like lodash or other utility packages?

File: src/components/usuarios/UsuarioForm.jsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    const _this = this;
+    const { firebase } = this.props;
+
+    // generate random password if empty
+    if (password === '' || password === undefined) {
+      password = randomString({length: 10});
+    }
+
+    // create user function url
+    const createUser = 'https://us-central1-igv-andamiaje-co.cloudfunctions.net/createUser';
```
Comment (varffs) [inline src/components/usuarios/UsuarioForm.jsx]: should this value be passed as a Global or something?

Comment (varffs) [reply src/components/usuarios/UsuarioForm.jsx]: I mean the base string before the function

File: src/components/usuarios/UsuarioForm.jsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        email: email,
+        password: password,
+      },
+    	headers: {
+        'Access-Control-Allow-Origin': '*',
+    	},
+      mode: 'no-cors'
+    })
+    .then((response) => {
+      // use set() to push to user profile
```
Comment (varffs) [inline src/components/usuarios/UsuarioForm.jsx]: Is this overwriting the account/user just made on the server or write a profile on the server separate from the user account in firebase?

**Review by varffs — CHANGES_REQUESTED** [review]
A few questions but also I think loading states need user feedback as the server functions take a little bit too long and it feels like nothing happened. Perhaps when disabled buttons are more styled that might help too. That includes the list view as I deleted something and it took quite a long time.

**Comment by varffs** [comment]
Ok def a bug to fix too: If you try to delete a user and keep trying again before the function returns you seem to be able to delete yourself. Must be some missing check on the delete function perhaps? Not too sure exactly why it happens but pretty dangerous

File: src/components/usuarios/UsuarioForm.jsx (hunk 2 above)
Comment (mikeyrayvon) [reply src/components/usuarios/UsuarioForm.jsx]: yes imported as constant

File: src/components/usuarios/UsuarioForm.jsx (hunk 3 above)
Comment (mikeyrayvon) [reply src/components/usuarios/UsuarioForm.jsx]: this writes a different record on the database with the same ID as the user created in firebase auth

**Pull request merged**